test/finally_test.cpp

```diff
@@ -27,12 +32,39 @@
 
 using namespace unifex;
 
-TEST(Finally, Pipeable) {
+TEST(Finally, Value) {
   timed_single_thread_context context;
 
-  schedule(context.get_scheduler())
-    | finally(schedule(context.get_scheduler()) 
-        | transform([](){ std::printf("finally\n"); }))
-    | transform([](){ std::printf("transform\n"); })
+  auto res = just(42)
+    | finally(schedule(context.get_scheduler()))
+    | transform([](int i){ return std::make_pair(i, std::this_thread::get_id() ); })
     | sync_wait();
+
+  ASSERT_FALSE(!res);
```

Comment:
I find the double negative less clear than a positive assertion.
```suggestion
  ASSERT_TRUE(res);
```
You can be more explicit, too:
```suggestion
  ASSERT_TRUE(res.has_value());
```